PodcastGenerator/core/feed_generator.php

```diff
@@ -43,17 +44,17 @@ function generateRSS()
         </itunes:owner>
         <itunes:explicit>' . $config['explicit_podcast'] . '</itunes:explicit>
 		<itunes:category text="' . htmlspecialchars($config['itunes_category[0]']) . '"></itunes:category>'."\n";
-    if ($config['itunes_category[1]'] != '') {
+    if (isset($config['itunes_category[1]']) && $config['itunes_category[1]'] != '') {
         $feedhead .= '		<itunes:category text="' . htmlspecialchars($config['itunes_category[1]']) . '"></itunes:category>'."\n";
     }
-    if ($config['itunes_category[2]'] != '') {
+    if (isset($config['itunes_category[1]']) && $config['itunes_category[2]'] != '') {
         $feedhead .= '		<itunes:category text="' . htmlspecialchars($config['itunes_category[2]']) . '"></itunes:category>'."\n";
     }
     // Get supported file extensions
     $supported_extensions = array();
     $supported_extensions_xml = simplexml_load_file($config['absoluteurl'] . 'components/supported_media/supported_media.xml');
-    foreach ($supported_extensions_xml->mediaFile->extension as $item) {
-        array_push($supported_extensions, $item);
+    foreach ($supported_extensions_xml->mediaFile as $item) {
+        array_push($supported_extensions, (string)$item->extension);
```

Comment:
Please use `strval()`, we use this rather than the C style cast

Reply:
that change will be made